src/libraries/System.Private.CoreLib/src/System/Globalization/CharUnicodeInfo.cs

```diff
@@ -250,6 +250,77 @@ private static double GetNumericValueNoBoundsCheck(uint codePoint)
             }
         }
 
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        internal static char ToUpper(char codePoint)
+        {
+            nuint offset = GetCategoryCasingTableOffsetNoBoundsChecks((uint)codePoint);
+
+            // The offset is specified in shorts:
+            // Get the 'ref short' corresponding to where the addend is, read it as a signed 16-bit value, then add
+
+            ref short rsStart = ref Unsafe.As<byte, short>(ref MemoryMarshal.GetReference(UppercaseValues));
+            ref short rsDelta = ref Unsafe.Add(ref rsStart, (nint)offset);
+            nint delta = (BitConverter.IsLittleEndian) ? rsDelta : BinaryPrimitives.ReverseEndianness(rsDelta);
+            return (char)(delta + (nint)codePoint);
+        }
+
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        internal static uint ToUpper(uint codePoint)
+        {
+            if (UnicodeUtility.IsBmpCodePoint(codePoint))
+            {
+                return ToUpper((char)codePoint);
+            }
+
+            if (!UnicodeUtility.IsValidCodePoint(codePoint))
+            {
+                ThrowHelper.ThrowArgumentOutOfRangeException(ExceptionArgument.codePoint);
+            }
+
+            nuint offset = GetCategoryCasingTableOffsetNoBoundsChecks(codePoint);
+
+            // The offset is specified in shorts:
+            // Get the 'ref short' corresponding to where the addend is, read it as a signed 16-bit value, then add
+
+            ref short rsStart = ref Unsafe.As<byte, short>(ref MemoryMarshal.GetReference(UppercaseValues));
+            ref short rsDelta = ref Unsafe.Add(ref rsStart, (nint)offset);
+            nint delta = (BitConverter.IsLittleEndian) ? rsDelta : BinaryPrimitives.ReverseEndianness(rsDelta);
```

Comment:
Similar comment as earlier: change this to read `int delta`, and change the statement below to read `return (uint)delta + codePoint`. (Both _delta_ and _codePoint_ are 32-bit integers going in to that addition, but _delta_ needs to be forcibly cast to an unsigned value to avoid the compiler wanting to widen everything to a 64-bit addition and messing with the codegen.)

Reply:
will do that.